httpconnect/httpconnect.go

```diff
@@ -147,28 +151,37 @@ func (f *HTTPConnectHandler) intercept(w http.ResponseWriter, req *http.Request)
 			}
 		}
 	}
-	var closeOnce sync.Once
-	go func() {
+
+	readErrCh := make(chan error)
+	op.Go(func() {
```

Comment:
So, this basically links the goroutine to this op, so if an error happens it gets reported as part of the op?